modules/apps/cookies/cookies-banner-web/src/main/java/com/liferay/cookies/banner/web/internal/servlet/taglib/EnableThirdPartyCookiesBottomJSDynamicInclude.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.cookies.banner.web.internal.servlet.taglib;
+
+import com.liferay.frontend.js.loader.modules.extender.npm.NPMResolver;
+import com.liferay.portal.kernel.servlet.taglib.DynamicInclude;
+import com.liferay.portal.kernel.servlet.taglib.aui.ScriptData;
+
+import java.io.IOException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Bryce Osterhaus
+ */
+@Component(service = DynamicInclude.class)
+public class EnableThirdPartyCookiesBottomJSDynamicInclude
+	implements DynamicInclude {
+
+	@Override
+	public void include(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse, String key)
+		throws IOException {
+
+		ScriptData scriptData = new ScriptData();
+
+		String initModuleName = _npmResolver.resolveModuleName(
```

Comment:
Can we migrate this project to ESM so that we don't need to care about npm resolutions, etc?

In fact, I would migrate every new functionality to ESM if possible...

Reply:
If you decide to do it you just need to pass a [JSFragment](https://github.com/liferay/liferay-portal/blob/53a8c5dd134b7cf6e6ff9efd9a97db1c629c901c/portal-kernel/src/com/liferay/portal/kernel/servlet/taglib/aui/JSFragment.java#L15) to ScriptData, instead of using the syntax you use below.

You just add it with [this method](https://github.com/liferay/liferay-portal/blob/53a8c5dd134b7cf6e6ff9efd9a97db1c629c901c/portal-kernel/src/com/liferay/portal/kernel/servlet/taglib/aui/ScriptData.java#L38).